shell/common/rasterizer.cc

```diff
@@ -343,10 +343,15 @@ RasterStatus Rasterizer::DrawToSurface(flutter::LayerTree& layer_tree) {
     if (raster_status == RasterStatus::kFailed) {
       return raster_status;
     }
-    frame->Submit();
     if (external_view_embedder != nullptr) {
-      external_view_embedder->SubmitFrame(surface_->GetContext());
+      external_view_embedder->SubmitFrame(surface_->GetContext(),
```

Comment:
We need to make sure the CATransaction is committed after everything is ready to be presented.
Right now `external_view_embedder->SubmitFrame` commits the CATransaction, so we can't move `frame->Submit` to be called after it.

Reply:
Maybe we can add a separate interface in `external_view_embedder` to commit transactions?

Reply:
Sounds like a good approach, however I'd probably call the method which (on iOS) commits the transaction `SubmitFrame` 😄.

What do you think about adding a `FinalizeCanvases` method which will do everything but commit the transaction, and then  separately `SubmitFrame`(which on iOS commits the transaction, on embedder platforms would present layers).

Reply:
I'm not too familiar with `embedder` platforms, does it mean iOS's `FinalizeCanvases` becomes an equivalent of embedder platforms' `SubmitFrame`?

Reply:
I added a `FinishFrame()` method to `ExternalViewEmbedder` that can be used for committing the transaction. I believe this is the simplest refactor given the scope of this PR.

Reply:
I'm OK with this only considering iOS. I'm not sure what the best approach would be considering external embedder. Probably need more inputs from @amirh or @chinmaygarde